src/sls_psi_api_connector.js

```diff
@@ -0,0 +1,254 @@
+ });
+ const auditKeys = Object.keys(lighthouseResult["audits"]);
+ const failedAuditNames = auditKeys.filter((auditName) => {
+  const score = lighthouseResult["audits"][auditName].score;
+  if (score < 1 && score != null) {
+    return auditName;
+  }
+ });
+
+ const crux = [];
+ if (loadingExperience["metrics"]) {
+   allResults.crux_data = true;
+   crux.push(loadingExperience["overall_category"]);
+   responseMap.get("crux").forEach((metricName) => {
+     if (loadingExperience["metrics"][metricName]) {
+       const metric = loadingExperience["metrics"][metricName];
+       let percentile = metric["percentile"];
+       if (metricName === "CUMULATIVE_LAYOUT_SHIFT_SCORE") {
+         percentile = percentile / 100;
+       }
+       crux.push(percentile);
+     } else {
+       crux.push("N/A");
+     }
+   });
+   // If there's insufficient field data for the page, the API responds with
+   // origin-level field data and origin_fallback = true.
+   if (loadingExperience["origin_fallback"]) {
+     allResults.origin_fallback = true;
+   }
+ }
+ allResults.data = [screenshot, ...crux, ...categories, ...metrics, failedAuditNames.toString(), version];
+ return allResults;
+}
+
+/**
+* Attaches an info note to the current last row of the sheet.
+*
+* @param {string} note The note to add.
+* @param {?string=} formatColor The background color of the note in rgb
+*     hex. The default null value leaves the color as is.
+*/
+function addNote(note, formatColor = null) {
+ const sheet = SpreadsheetApp.getActive().getSheetByName(RESULTS_TAB);
+ const lastRow = sheet.getLastRow();
+ sheet.getRange(`${lastRow}:${lastRow}`).setBackground(formatColor);
+ sheet.getRange(`D${lastRow}`).setNote(note);
+}
+
+/**
+* Creates a results map for the given row of the URL settings array.
+// These are used to index the objects returned from PSI,
+// which is why they are named as they are. The order they are defined here
+// is also the order they are inserted into the sheet per parseResults.
+*
+* @return {!Map<string, !Map<string, number>>} The budget values in an object.
+*/
+function createResultsMap() {
+  const categories = ["performance"];
+  const crux = [
+    "FIRST_CONTENTFUL_PAINT_MS",
+    "LARGEST_CONTENTFUL_PAINT_MS",
+    "FIRST_INPUT_DELAY_MS",
+    "CUMULATIVE_LAYOUT_SHIFT_SCORE",
+    "INTERACTION_TO_NEXT_PAINT",
+  ];
+  const metrics = [
+    "server-response-time",
+    "first-contentful-paint",
+    "largest-contentful-paint",
+    "total-blocking-time",
+    "cumulative-layout-shift"
+  ];
+  const assets = [
+    "total",
+    "script",
+    "image",
+    "stylesheet",
+    "document",
+    "font",
+    "other",
+    "media",
+    "third-party",
+  ];
+
+
+  const requiredResultsMap = new Map();
+  requiredResultsMap.set("categories", categories);
+  requiredResultsMap.set("metrics", metrics);
+  requiredResultsMap.set("assets", assets);
+  requiredResultsMap.set("crux", crux);
+  return requiredResultsMap;
+}
+
+//  audits = [];
```

Comment:
This was commented out because it wasn't used. This version of the PSI API connector deviates from Xavier's version. Will delete the commented out bit.